internal/bminventory/inventory.go

```diff
@@ -2872,6 +2872,9 @@ func (b *bareMetalInventory) V2DeregisterHostInternal(ctx context.Context, param
 		hostutil.GetHostnameForMsg(&h.Host))
 
 	if h.ClusterID != nil {
+		if _, err = b.refreshClusterStatus(ctx, h.ClusterID, b.db); err != nil {
```

Comment:
Won't that make the flow synchronous?
Did you check how long it takes for `refreshClusterStatus` to run?

Reply:
This was exactly the intention of this PR. And yes this solution was approved. The Ux experience in SaaS takes precedence here.